### ISSUE TITLE
Can't use FSharp.Interop.Dynamic in Azure Functions

### ISSUE THREAD

**Issue by ulrikaugustsson** [Azure/azure-webjobs-sdk-script#861]
Using FSharp.Interop.Dynamic gives a error that it can't find Dynamitey.
I have followed the steps here: https://github.com/Azure/azure-content/blob/master/articles/azure-functions/functions-bindings-documentdb.md
And found another person that recently had the same issue as me, that link is under repro.

#### Repro steps

Provide the steps required to reproduce the problem

http://stackoverflow.com/questions/40294107/dynamitey-exception-when-accessing-input-documentdb-with-f-in-azure-functions

#### Expected behavior

The function should work without exception.

#### Actual behavior

It gives an exception:

```
Exception while executing function: Functions.myFunctionName. mscorlib: Exception has been thrown by the target of an invocation. FSharp.Interop.Dynamic: Could not load file or assembly 'Dynamitey, Version=1.0.2.0, Culture=neutral, PublicKeyToken=cbf53ea3aeb972c6' or one of its dependencies. The system cannot find the file specified.
```

#### Related information 

Using F#

**Comment by lindydonna** [comment]
@dsyme and @sylvanc, could you please investigate

**Comment by dsyme** [comment]
@ulrikaugustsson   @lindydonna @fabiocav @christopheranderson - @sylvanc and I are actively looking into this bug this week. We think nearly all the existing F# issues boil down to the one same bug (not finding assemblies at runtime which have been resolved statically)

Suspected duplicate of #650, but we need to investigate further.

**Comment by ulrikaugustsson** [comment]
@dsyme it might be interesting to note that for me it compiles fine but when I call the function I get the error. So it does not seem to be exactly the same as #650.

**Comment by dsyme** [comment]
@ulrikaugustsson Perhaps https://github.com/Azure/azure-webjobs-sdk-script/issues/819 is a clearer ikely duplicate?  Regardless, I'm trying in #891 to add a specific new scenario-like test for this

**Comment by mariomeyrelles** [comment]
I tried with this simplified code, that yields the same problem. Also, I tried to workaround of moving all files /bin folder with no success:

```
//----------------------------------------------------------------------------------------
// This prelude allows scripts to be edited in Visual Studio or another F# editing environment 

#if !COMPILED
#I @"../../bin/Binaries/WebJobs.Script.Host"
#r "Microsoft.Azure.WebJobs.Host.dll"
#r "System.Web.Http.dll"
#r "System.Net.Http.dll"


#endif

#r "../bin/Dynamitey.dll"
#r "../bin/FSharp.Interop.Dynamic.dll"

//----------------------------------------------------------------------------------------
// This is the body of the function 

open System
open System.Linq
open System.Net
open System.Net.Http
open System.Threading.Tasks
open Microsoft.Azure.WebJobs.Host
open FSharp.Interop.Dynamic
open Dynamitey

let Run (req: HttpRequestMessage , log: TraceWriter)  =
        
        let queryParams = req.GetQueryNameValuePairs().ToDictionary((fun p -> p.Key), (fun p -> p.Value), StringComparer.OrdinalIgnoreCase)
        let nameInsideBodyOfRequest = req.Content?name //Causes the error

        log.Verbose(sprintf "F# HTTP trigger function processed a request. Name=%A" req.RequestUri)

        let res =
            match queryParams.TryGetValue("name") with 
            | true, name -> 
                new HttpResponseMessage(HttpStatusCode.OK, Content = new StringContent("Hello " + name))
            | _ -> 
                new HttpResponseMessage(HttpStatusCode.BadRequest, Content = new StringContent("Please pass a name on the query string"))

        res

```

**Comment by fabiocav** [comment]
Resolved by #893
